pilot/pkg/xds/ep_filters.go

```diff
@@ -67,6 +67,13 @@ func (b *EndpointBuilder) EndpointsByNetworkFilter(endpoints []*LocLbEndpointsAn
 
 		// Process all of the endpoints.
 		for i, lbEp := range ep.llbEndpoints.LbEndpoints {
+			istioEndpoint := ep.istioEndpoints[i]
+
+			// If the proxy can't view the network for this endpoint, exclude it entirely.
+			if !b.proxyView.IsVisible(istioEndpoint) {
```

Comment:
This seems correct -- but also changes L80-86 to not mutate the weight.

Is this fixing a bug? Seems like it might be ... Or somehow it has no impact?

Reply:
My thought as well. There hasn't been a bug filed that I'm aware of ... and even if there was a bug, it would likely not be very noticeable. However, I believe this code is (more) correct.